mods/ra/maps/allies-09a/rules.yaml

```diff
@@ -0,0 +1,129 @@
+Player:
+	PlayerResources:
+		DefaultCash: 10000
+
+World:
+	LuaScript:
+		Scripts: campaign-global.lua, allies09a.lua, allies09a-AI.lua
+	MissionData:
+		Briefing: One of Stalin's top atomic strategists, Vladimir Kosygin, wishes to defect. His knowledge of Stalin's atomic strategies is invaluable to us. We will extract him from the Riga compound where he is stationed.\n\nUse a spy to infiltrate the Soviet command center and contact Kosygin. Once he is out of the building, guide him back to your base any way you can.
+		StartVideo: 
+		WinVideo: 
+		LossVideo: 
+	ScriptLobbyDropdown@difficulty:
+		ID: difficulty
+		Label: Difficulty
+		Values:
+			easy: Easy
+			normal: Normal
+			hard: Hard
+		Default: normal
+
+LST.Reinforcement:
+	Inherits: LST
+	RejectsOrders:
+	-Buildable:
+	-Selectable:
+	RenderSprites:
+		Image: lst
+	Interactable:
+
+powerproxy.paratroopers:
+	ParatroopersPower:
+		DropItems: E1,E1,E1,E2,E2
+
+MCV:
+	Buildable:
+		Prerequisites: ~disabled
+
+TRUK:
+	Buildable:
+		Prerequisites: ~disabled
+
+MRJ:
+	Buildable:
+		Prerequisites: ~disabled
+
+3TNK:
+	Buildable:
+		Prerequisites: ~vehicles.soviet
+
+V2RL:
+	Buildable:
+		Prerequisites: ~vehicles.soviet
+
+4TNK:
+	Buildable:
+		Prerequisites: ~vehicles.soviet
+
+QTNK:
+	Buildable:
+		Prerequisites: ~disabled
+
+AFLD:
+	Buildable:
+		Prerequisites: ~disabled
+
+HPAD:
+	Buildable:
+		Prerequisites: ~disabled
+
+ATEK:
+	Buildable:
+		Prerequisites: ~disabled
+
+STEK:
+	Buildable:
+		Prerequisites: ~disabled
+
+PDOX:
+	Buildable:
+		Prerequisites: ~disabled
+
+IRON:
+	Buildable:
+		Prerequisites: ~disabled
+
+MSLO:
+	Buildable:
+		Prerequisites: ~disabled
+
+GAP:
+	Buildable:
+		Prerequisites: ~disabled
+
+BRIK:
+	Buildable:
+		Prerequisites: ~disabled
+
+MECH:
+	Buildable:
+		Prerequisites: ~disabled
+
+THF:
+	Buildable:
+		Prerequisites: ~disabled
+
+E7:
+	Buildable:
+		Prerequisites: ~disabled
+
+E7.noautotarget:
+	Buildable:
+		Prerequisites: ~disabled
+
+MIG:
+	Buildable:
+		Prerequisites: ~disabled
+
+CA:
+	Buildable:
+		Prerequisites: ~disabled
+
+MSUB:
+	Buildable:
+		Prerequisites: ~disabled
+
+FCOM:
+	Targetable:
+		TargetTypes: Ground, Structure, C4, DetonateAttack, SpyInfiltrate
```

Comment:
This needs to use `GroundActor` instead of `Ground`.

Reply:
Substitute Ground to GroundActor

Reply:
Hi @abcdefg30,

I try to set new TargetTypes name for GNRL(Kosygin) in the rules.yaml. After set, Kosygin seems invincible.
Seems dogs need new ValidTargets. But after I set the ValidTargets under DogJaw (also under rules.yaml).
I cannot launch this map.

Maybe I miss something. Could you give me some advice?

GNRL:
	Targetable:
		TargetTypes: Kosygin
	ProximityCaptor:
		Types: Kosygin

DogJaw:
	ValidTargets: Infantry, Kosygin
	Warhead@1Dam: TargetDamage
		Damage: 100000
		ValidTargets: Infantry, Kosygin

Reply:
You don't need `ProximityCaptor`. Without seeing the yaml I can't tell much more. You should run the `--check-yaml` command and look at the output. (E.g. run `OpenRA.Utility.exe ra --check-yaml ./mods/ra/maps/allies-09a` from the command line in the OpenRA directory on Windows.)

Reply:
Ah wait, `DogJaw` is a weapon, so you can't override it in `rules.yaml`. You need to define a new `weapons.yaml` and do it there. Then add `Weapons: weapons.yaml` to `map.yaml` like here: https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/maps/survival02/map.yaml#L1012

Reply:
Hi @abcdefg30 

After add new weapons.yaml. Only dogs can attack Kosygin. Thanks.

Clarify PrimaryObjective when Kosygin contacted.
Can't select extraction helicopter.
More script to let this map run like the original.